### ISSUE TITLE
Fix tests.foreman.ui.test_docker.DockerContainersTestCase.test_create_container_compute_resource

### ISSUE THREAD

**Issue by oshtaier** [SatelliteQE/robottelo#3044]
The next issue is reproduced on CI and needs to be analyzed and fixed:

```
Error Message

Container with name FkSvCix5aq was not created successfully

**Stacktrace**
File "/home/jenkins/shiningpanda/jobs/ca586003/virtualenvs/d41d8cd9/lib/python2.7/site-packages/unittest2/case.py", line 67, in testPartExecutor
    yield
  File "/home/jenkins/shiningpanda/jobs/ca586003/virtualenvs/d41d8cd9/lib/python2.7/site-packages/unittest2/case.py", line 625, in run
    testMethod()
  File "robottelo/decorators.py", line 136, in wrapper
    return func(*args, **kwargs)
  File "/home/jenkins/workspace/satellite6-automation-downstream-rhel71-ui/tests/foreman/ui/test_docker.py", line 1213, in test_create_container_compute_resource
    parameter_list=self.parameter_list,
  File "robottelo/ui/factory.py", line 700, in make_container
    Container(session.browser).create(**create_args)
  File "robottelo/ui/container.py", line 107, in create
    .format(name)
```

**Comment by abalakh** [comment]
Together with @oshtaier investigated the issue and it seems we've faced issue similar to [BZ1253270](https://bugzilla.redhat.com/show_bug.cgi?id=1253270) - clicking 'Submit' button on creation page doesn't do anything.

production.log contains just 1 line:
```2015-11-16 04:47:56 [I] Failed to save: ```

I've already submitted a new BZ for that - [BZ1282431](https://bugzilla.redhat.com/show_bug.cgi?id=1282431).
Some outcomes of our investigation:
- Docker container can't be created with default provided docker package (`docker-1.8.2-7.el7.x86_64`)
- Downgrading docker package to `docker-1.7.1-115.el7.x86_64` resolves the issue
- Downgrading other packages like `ruby193-rubygem-foreman_docker` seem to have no effect

Also would like to note that the test is failing starting from Nov 3, so it's not an issue with specific and/or latest compose.

**Comment by abalakh** [comment]
One more point - testing docker containers requires `katello-ca` certificate to be installed on sat6 machine, so we need to run
```rpm -Uvh http://{hostname}/pub/katello-ca-consumer-latest.noarch.rpm```
before running docker containers tests, otherwise test will continue to fail.
IMO, the best place to integrate certificate installation will be automation tools, so we may think of creating a new issue there.

**Comment by elyezer** [comment]
@abalakh what about installing it on setUpClass?

**Comment by abalakh** [comment]
@elyezer not sure if it's a better idea. 3 main drawbacks of checking the certificate in setUpClass:
- we'll need to put such verification into setUpClass of each class which will ever use docker containers. For now it's UI docker module for sure, and possibly CLI and API tests as well. And each docker test module has 7 classes, so in case we need containers somewhere outside `DockerContainersTestCase`, we'll need to copy setUpClass there, too. And what if we need to update smoke tests to check docker tests? I'm just afraid we can introduce a lot of duplicated code, and creating helper func won't fully fix a problem - we'll still need to add setUpClass with helper function call everywhere.
- we'll check for certificate presence each time test suite will be executed. so if i wanna run some test 20 times, robottelo will check the rpm 20 times too. It shouldn't be pretty time consuming, but redundant for sure.
- if ssh command fails(e.g. connection issue), whole test class will be marked as failed, which is not good especially when we have 20 tests in class 19 of which don't use docker container

Fixing it on automation tools level won't introduce extra code at all, won't introduce any extra checks for each test suite and can't accidentally fail whole class. Furthermore, it can be useful for manual testing as well - even if you aren't planning using robottelo but want to use docker container functionality, you'll still need to install certificate, so imo optional checkbox for installer job would be pretty handy.

All the mentioned points are pretty weak actually, so if you have another vision please share your thoughts, for me setUpClass doesn't sound like something unacceptable :)

**Comment by oshtaier** [comment]
@elyezer I don't like the idea to put code into robottelo either. We have a lot of discussions with @abalakh and we generate reasons above, but as Andrii said it is possible that putting something to auto-tools is also bad idea.

Probably, one more item from my side:
It is very nice to avoid putting more CLI code into UI module if we can as we will prevent collapsing architecture more and more further

**Comment by sthirugn** [comment]
Couple of things:
* How did the container create test cases work before without registering satellite to itself?
* Are we proposing to register the satellite machine to itself? This will remove cdn server info from the satellite.  As a result, this will fail any sync tests - which includes any test which syncs sat6tools from cdn, longrun tests which syncs rhel repos from cdn, etc.

**Comment by oshtaier** [comment]
@sthirugn 

- How did the container create test cases work before without registering satellite to itself?

They didn't work before as test cases were executed only locally where corresponding certificate was installed. Basically, that fact was forgotten in a long list of other things.
Also, of course, we need it only for UI, so all other areas (API/CLI) will work as intended.
At that moment, we have only one fail for that functionality on CI as all other tests are wrapped in BZ decorators, so not only create procedure, but all other cases will fail once corresponding bugs will be fixed by developers.

- Are we proposing to register the satellite machine to itself? This will remove cdn server info from the satellite. As a result, this will fail any sync tests - which includes any test which syncs sat6tools from cdn, longrun tests which syncs rhel repos from cdn, etc.

Yes and yes. You absolutely right, so we need definitely think on any other alternative. Maybe, we need to create new VM with docker or something. At that moment, we do not have ready solution

**Comment by abalakh** [comment]
So fixing it on automation-tools level is unacceptable.
> Are we proposing to register the satellite machine to itself? This will remove cdn server info from the satellite. As a result, this will fail any sync tests - which includes any test which syncs sat6tools from cdn, longrun tests which syncs rhel repos from cdn, etc.

Does this mean sync tests will fail only while machine is registered to itself or unregistering won't fix the issue?
If unregistering resolves that, then we could still try to install the rpm in `setUp` section and remove it in `tearDown` section.
If unregistering doesn't resolves that issue, then I guess we're doing something wrong. It sounds weird for me that we have to choose in either sync or docker testing.

**Comment by oshtaier** [comment]
@abalakh we can test that out, but as far as I remember, unregister procedure should help

**Comment by sthirugn** [comment]
`Does this mean sync tests will fail only while machine is registered to itself or unregistering won't fix the issue?'
Nope. Unregistering won't fix this.  You need to clean up the red hat subscription management server things as shown in https://github.com/SatelliteQE/automation-tools/blob/master/automation_tools/__init__.py#L1284

**Comment by oshtaier** [comment]
Wow, so it means that I don't know what I am talking about. Thanks @sthirugn for pointing out on that fact

**Comment by oshtaier** [comment]
@sthirugn btw, maybe you have any idea how to resolve that issue? That will help us a lot as that area is pretty unknown for us

**Comment by omaciel** [comment]
Hey guys, how about performing the installation/verification of `katello-ca` as a `setUpModule` in the modules that need them? This would be called only once for all tests in the module:

```python
def setUpModule():
    install_katello_ca()

def tearDownModule():
    remove_katello_ca()
```

**Comment by oshtaier** [comment]
It seems that `remove_katello_ca()` will not be so trivial task according to @sthirugn code example, but anyway thanks for ideas

**Comment by sthirugn** [comment]
@oshtaier: yes +1 to @omaciel's idea.

**Comment by omaciel** [comment]
@oshtaier @sthirugn actually, it is trivial to do it, since we already have a task in `automation-tools` that can handle it... but I'd even vote to **not** remove it. It doesn't hurt us at all.

**Comment by sthirugn** [comment]
@omaciel `but I'd even vote to not remove it` - do you mean not removing katello-cert and not resetting rhsm? This will affect future tests as some tests may sync tools repo or something else from cdn.

**Comment by sthirugn** [comment]
@omaciel you may be right, does it totally depend on the satellite specified cdn url?

**Comment by sthirugn** [comment]
@oshtaier, @abalakh : @omaciel is right.  Since we are not getting sat tools in the sat host itself.  But we do it from the satellite application which is configured with cdn info already.  Registering the satellite host to itself should not be a problem for the sync tests.

**Comment by omaciel** [comment]
@sthirugn to be safe, we can remove `katello-ca`. I really do think it should be trivial to do so.

**Comment by oshtaier** [comment]
@omaciel @sthirugn Huge thanks to you guys for all these inputs

**Comment by omaciel** [comment]
@oshtaier thank you too and @abalakh for the research and time spent on this.
